lib/publiccloud/gce.pm

```diff
@@ -81,8 +83,66 @@ sub upload_img {
     return $img_name;
 }
 
+sub ipa {
+    my ($self, %args) = @_;
+
+    $args{instance_type}        //= 'n1-standard-2';
+    $args{cleanup}              //= 1;
+    $args{ssh_private_key_file} //= '.ssh/id_rsa';
+    $args{tests}                //= '';
+    $args{timeout}              //= 60 * 20;
+    $args{results_dir}          //= 'ipa_results';
+
+    $args{tests} =~ s/,/ /g;
+
+    if (script_run('test -f ' . $args{ssh_private_key_file}) != 0) {
+        assert_script_run('SSH_DIR=`dirname ' . $args{ssh_private_key_file} . '`; test -d $SSH_DIR || mkdir -p $SSH_DIR');
+        assert_script_run('ssh-keygen -b 2048 -t rsa -q -N "" -f ' . $args{ssh_private_key_file});
+    }
+
+    my $cmd = 'ipa --no-color test gce ';
+    $cmd .= '--debug ';
+    $cmd .= '--service-account-file "' . $credentials_file . '" ';
+    $cmd .= '--distro sles ';
+    $cmd .= '--ssh-private-key-file "' . $args{ssh_private_key_file} . '" ';
+    $cmd .= '--region "' . $self->region . '" ';
+    $cmd .= '--results-dir "' . $args{results_dir} . '" ';
+    $cmd .= ($args{cleanup}) ? '--cleanup ' : '--no-cleanup ';
+    $cmd .= '--instance-type "' . $args{instance_type} . '" ';
+
+    if (exists($args{running_instance_id})) {
+        $cmd .= '--running-instance-id "' . $args{running_instance_id} . '" ';
+    } else {
+        $cmd .= '--image-id "' . $args{image_id} . '" ';
+    }
+    $cmd .= $args{tests};
+
+    my $output = script_output($cmd . ' 2>&1', $args{timeout}, proceed_on_failure => 1);
+    my $ipa = $self->parse_ipa_output($output);
+    die($output) unless (defined($ipa));
+
+    # retrieves username and password for ssh login
+    $ipa->{username} = 'susetest';
+    $ipa->{ssh_key}  = $args{ssh_private_key_file};
+
+    $self->{running_instances} //= {};
+    if ($args{cleanup}) {
+        delete($self->{running_instances}->{$ipa->{instance_id}});
+    } else {
+        $self->{running_instances}->{$ipa->{instance_id}} = $ipa;
+    }
+
+    return $ipa;
+}
+
 sub cleanup {
-    record_info('info', 'to be implemented');
+    my ($self) = @_;
+
+    print Dumper($self->{running_instances});
```

Comment:
Seems to be a left over for debugging, also in `azure.pm`. I think we can remove it.